digest/src/core_api/ct_variable.rs

```diff
@@ -0,0 +1,74 @@
+use super::{AlgorithmName, FixedOutputCore, UpdateCore, VariableOutputCore};
+use block_buffer::BlockBuffer;
+use core::{fmt, marker::PhantomData};
+use generic_array::typenum::type_operators::IsLessOrEqual;
+use generic_array::{ArrayLength, GenericArray};
+
+/// Wrapper around [`VariableOutputCore`] which selects output size
+/// at compile time.
+#[derive(Clone)]
+pub struct CtVariableWrapper<T, OutSize>
+where
+    T: VariableOutputCore,
+    OutSize: ArrayLength<u8> + IsLessOrEqual<T::MaxOutputSize>,
```

Comment:
Unfortunately `IsLessOrEqual` will be not an easy bound to specify using current design of const generics.

Reply:
Welp, looks `IsLessOrEqual` bound does nothing, at least when used with type aliases...

Reply:
Hm, looks like I was using it incorrectly. The bound should look like this:
```rust
OutSize: ArrayLength<u8> + IsLessOrEqual<T::MaxOutputSize>,
LeEq<OutSIze, T::MaxOutputSize>: NonZero
```

Reply:
The bound is now fixed, but type aliases still do not check bounds, so compilation error will not happen until an incorrect alias will be used in code. Not ideal, but we can not do anything about it.